### PR TITLE
Guardian Spirits/Parasitic Nanomachine Swarm/Scarab Hive

### PR DESCRIPTION
**Pull request by ghost**
Adds a new Guardian mob, with three different fluff variants (magic/tech/bio).

When summoned/created, the mob is stored inside the summoner, and can manifest (hide back inside)/recall itself, as well as communicate privately via verb. The summoner can speak privately with the mob as well.

The mob itself is invincible, but can't live without a summoner/host, and will be warped back to them if they try to move too far away. In addition, depending on which ability set the mob has, it will deal a percentage of damage it suffers from attacks back to its host.

There are 8 variants with various quirks/powers, such as igniting enemies on touch, healing on attack, ranged attacks, etc

Added a creation tool for them, with magic/tech/bio variants.

Needs sprites for all three mob variants, as well as for the summoning items, and ideally more interesting powers later, but I think they're finished enough for playtesting (currently adminspawn only).

UPDATE: 

You can now edit which types are selectable by the spawner, and whether the spawner is random or input.

Also fixed the fluff strings the way Remie wanted them.

### CODE REVIEW

**Comment by RemieRichards** [comment]
add the GODMODE status flag to them to make them REALLY invincible.

**Comment by daman997** [comment]
so will this be added to wiz's or something later or will it just be a badmin tool forever

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+
+
+/obj/item/weapon/guardiancreator/biological
+	name = "scarab egg cluster"
+	desc = "A parasitic species that will nest in the closest living creature upon birth. While not great for your health, they'll defend their new 'hive' to the death."
+	icon = 'icons/obj/syringe.dmi'
+	icon_state = "combat_hypo"
+	theme = "bio"
+	mob_name = "Scarab Swarm"
+	use_message = "The egs begin to twitch..."
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: egs

**Comment by Jalleo** [comment]
Sounds like this can be for wiz syndie and changeling

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 2]
```diff
@@ -0,0 +1,400 @@
...
+	var/gaurdiantype = pick("Fire", "Standard", "Scout", "Shield", "Ranged", "Healer", "Fast")
+	var/pickedtype = /mob/living/simple_animal/hostile/guardian/punch
+	var/picked_color = randomColor(0)
+	switch(gaurdiantype)
+
+		if("Fire")
+			pickedtype = /mob/living/simple_animal/hostile/guardian/fire
+			switch (theme)
+				if("magic")
+					user << "..And draw Atmosia, bringer of cleansing fires!"
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: can you please put these strings on the mobs?
eg:
var/magic_fluff_string
var/tech_fluff_string
var/bio_fluff_string

and then you can do ```user << initial(pickedype.magic_fluff_string)``` (you'll need to replace var/pickedtype with ```var/mob/living/simple_animal/hostile/guardian/pickedtype```

**Comment by KorPhaeron** [comment]
@RemieRichards 
Would that fuck up the way I've done their AdjustBruteLoss?

@daman997

Remains to be seen. Could potentially work as a ling item (scarabs) or traitor/RD/nuke (nanomachines), or yeah, wizard.

Depends on how strong they end up being, how in depth they end up being, etc. Right now its all very basic and (I hope) erring on underpowered.

I could make a wizard only variant of the spawner that summons stronger versions with full spells, like fireball, timestop, lightning, etc.

**Comment by RemieRichards** [comment]
@KorPhaeron actually it wouldn't because your AdjustBruteLoss doesn't call ..() so GODMODE won't work anyway.

tl;dr your code is fine, ignore GODMODE.

**Comment by Anonus** [comment]
IIRC godmode doesn't work properly on simple animals anyway.

**Comment by RemieRichards** [comment]
@Anonus Phil fixed that, iirc.

**Comment by Anonus** [comment]
There should be a way to set a spawner to spawn only a specific type instead of it being random, for, say, cult or something.

**Comment by KorPhaeron** [comment]
@Anonus 

I could have a var/random, and it either letting you pick via input or choose randomly based on that.

**Comment by Anonus** [comment]
There should be two pick types, one being random and one being from an editable list(so that you could, say, make one that only ever spawned the fire guardian)

**Comment by dannno** [comment]
i have been waiting for this day

**Comment by Shadowlight213** [comment]
As much as I love new things being given to R&D, I think this shouldn't be made something that could be mass produced. I'd say keep it out of R&D. They have too much cool shit already.

**Comment by AnturK** [comment]
Alright, this might sound silly but : Will moving as the guardian inside summoner makes them burst ? Because i'm pretty sure that's default action for living mobs in other living mobs.

**Comment by Durkel** [comment]
Looks like it's time to stand proud.

**Comment by TheVekter** [comment]
So they're Stands.

**Comment by KorPhaeron** [comment]
@RemieRichards 

Since it's all self contained adminspawn only code and won't break/unbalance anything else, any chance of this getting merged tonight (if you get a chance to review the new stuff) so I can test it a couple times before I leave on Friday?

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 3]
```diff
@@ -0,0 +1,526 @@
...
+	var/tech_fluff_string = "BOOT SEQUENCE COMPLETE. ERROR MODULE LOADED. THIS SHOULDN'T HAPPEN. Submit a bug report!"
+	var/bio_fluff_string = "Your scarabs fail to mutate. This shouldn't happen! Submit a bug report!"
+
+
+/mob/living/simple_animal/hostile/guardian/Life() //Dies if the summoner dies
+	..()
+	if(summoner)
+		if(summoner.stat == DEAD)
+			src << "<span class='danger'>Your summoner has died!</span>"
+			visible_message("<span class='danger'>The [src] dies along with it's user!</span>")
```
Comment (Anonus) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: its

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 4]
```diff
@@ -0,0 +1,526 @@
...
+			ghostize()
+			qdel(src)
+	else
+		src << "<span class='danger'>Your summoner has died!</span>"
+		visible_message("<span class='danger'>The [src] dies along with it's user!</span>")
```
Comment (Anonus) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: its

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 5]
```diff
@@ -0,0 +1,526 @@
...
+	var/mob/living/spawner
+
+
+
+/obj/item/weapon/guardian_bomb/proc/disguise(var/obj/A)
+	A.loc = src
+	stored_obj = A
+	name = A.name
+	desc = A.desc
+	icon = A.icon
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: replace:
icon
icon_state
overlays
and I believe the pixel_X/Ys

with appearance = A.appearance

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 6]
```diff
@@ -0,0 +1,526 @@
...
+	icon_state = A.icon_state
+	overlays = A.overlays
+	anchored = A.anchored
+	density = A.density
+	pixel_y = A.pixel_y
+	pixel_x = A.pixel_x
+	spawn(600)
+		stored_obj.loc = get_turf(src.loc)
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: ```stored_obj``` might not exist anymore
```spawner``` might not exist anymore
```src``` might not exist anymore

600 ticks is 60 seconds, a lot can happen in a minute.

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 7]
```diff
@@ -0,0 +1,526 @@
...
+	tech_fluff_string = "Boot sequence complete. Standard combat modules loaded. Nanoswarm online."
+	bio_fluff_string = "Your scarab swarm stirs to life, ready to tear apart your enemies."
+	var/battlecry = "AT"
+
+/mob/living/simple_animal/hostile/guardian/punch/verb/Battlecry()
+	set name = "Set Battlecry"
+	set category = "Guardian"
+	set desc = "Choose what you shout as you punch"
+	var/input = stripped_input(src,"What do you want your battlecry to be? Max length of 6 characters.", ,"", 6)
+	src.battlecry = input
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: please check input is not an empty string first.

File: code/modules/mob/living/simple_animal/guardian/guardian.dm [hunk 8]
```diff
@@ -0,0 +1,526 @@
...
+	var/damage = amount * src.damage_transfer
+	if (src.summoner)
+		src.summoner.adjustBruteLoss(damage)
+		if(damage)
+			src.summoner << "<span class='danger'><B>Your [src.name] is under attack! You take damage!</span></B>"
+
+
+/mob/living/simple_animal/hostile/guardian/ex_act(severity, target)
+	switch (severity)
+		if (1.0)
```
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/guardian/guardian.dm]: remove the ```.0``` it's found in other procs because it's a decompilation artefact of ```1``` --> ```1.0```
